python_modules/dagster/dagster/_core/definitions/asset_spec.py

```diff
@@ -386,6 +387,36 @@ def merge_attributes(
                 partitions_def=self.partitions_def,
             )
 
+    def with_attributes(
```

Comment:
we already have a top level `replace_attributes` that you should be able to reuse for this.

Reply:
thanks! saw that but didn't know that was the rec. will update.